subsys/bluetooth/host/conn.c

```diff
@@ -1214,41 +1296,62 @@ static void conn_cleanup(struct bt_conn *conn)
 	k_delayed_work_submit(&conn->update_work, K_NO_WAIT);
 }
 
+static int conn_prepare_events(struct bt_conn *conn,
+			       struct k_poll_event *events)
+{
+	if (!atomic_get(&conn->ref)) {
+		return -ENOTCONN;
+	}
+
+	if (conn->state == BT_CONN_DISCONNECTED &&
+	    atomic_test_and_clear_bit(conn->flags, BT_CONN_CLEANUP)) {
+		conn_cleanup(conn);
+		return -ENOTCONN;
+	}
+
+	if (conn->state != BT_CONN_CONNECTED) {
+		return -ENOTCONN;
+	}
+
+	BT_DBG("Adding conn %p to poll list", conn);
+
+	k_poll_event_init(&events[0],
```

Comment:
could be just `events` here.

and `event->tag` further below.

Reply:
Right, that is something that we can indeed improve latter on.